src/plugins/analysis/cve_lookup/internal/database/db_setup.py

```diff
@@ -0,0 +1,116 @@
+import re
+import sys
+from pathlib import Path
+
+try:
+    from ..database.db_connection import DbConnection
+    from ..database.schema import Association, Cve, Cpe
+except (ImportError, SystemError):
+    sys.path.append(str(Path(__file__).parent.parent / 'database'))
+    from db_connection import DbConnection
+    from schema import Association, Cve, Cpe
+
+try:
+    from ..internal.helper_functions import CveEntry, replace_wildcards
+except (ImportError, SystemError):
+    sys.path.append(str(Path(__file__).parent.parent.parent / 'internal'))
+    from helper_functions import CveEntry, replace_wildcards
+
+CPE_SPLIT_REGEX = re.compile(r'(?<![\\:]):(?!:)|(?<=\\:):')  # don't split on '::' or '\:' but split on '\::'
+
+
+class DbSetup:
+    DB_PATH = str(Path(__file__).parent / 'cve_cpe.db')
+
+    def __init__(self, connection: DbConnection):
+        self.connection = connection
+        self.connection.create_tables()
+        self.session = self.connection.create_session()
+
+    def create_cve(self, cve_item: CveEntry) -> Cve:
+        '''
+        Create a Cve object from a CveEntry object.
+        '''
+        year = cve_item.cve_id.split('-')[1]
+        score_v2 = cve_item.impact.get('cvssMetricV2', 'N/A')
+        score_v30 = cve_item.impact.get('cvssMetricV30', 'N/A')
+        score_v3 = cve_item.impact.get('cvssMetricV31', score_v30)
+        return Cve(
+            cve_id=cve_item.cve_id,
+            year=year,
+            summary=cve_item.summary,
+            cvss_v2_score=score_v2,
+            cvss_v3_score=score_v3,
+        )
+
+    def create_cpe(self, cpe_id: str) -> Cpe:
+        '''
+        Create a Cpe object from a CPE ID.
+        '''
+        cpe_elements = replace_wildcards(CPE_SPLIT_REGEX.split(cpe_id)[2:])
+        (
+            part,
+            vendor,
+            product,
+            version,
+            update,
+            edition,
+            language,
+            sw_edition,
+            target_sw,
+            target_hw,
+            other,
+        ) = cpe_elements
+        return Cpe(
+            cpe_id=cpe_id,
+            part=part,
+            vendor=vendor,
+            product=product,
+            version=version,
+            update=update,
+            edition=edition,
+            language=language,
+            sw_edition=sw_edition,
+            target_sw=target_sw,
+            target_hw=target_hw,
+            other=other,
+        )
+
+    def add_cve_items(self, cve_list: list[CveEntry]):
+        '''
+        Add CVE items to the database.
+        '''
+        existing_cve_ids = {cve.cve_id for cve in self.session.query(Cve.cve_id).all()}
+        existing_cpe_ids = {cpe.cpe_id for cpe in self.session.query(Cpe.cpe_id).all()}
```

Comment:
Was the idea to only add new entries for efficiency? There is one problem with this approach: Many existing CVE entries get updated and this approach would ignore the updates. IMHO this is fast enough that we could just create a new DB with every installation. On the other hand, it would also be easy to do this by deleting the existing DB file an rerunning the installation.

Reply:
Yes the idea was to add only new entries. I didn't know that old CVE entries could also change. I've changed `install.py` to drop all tables before filling the Database again and I removed the update feature from `db_setup.py`.